RELEASE/scripts/autoscend/quests/level_any.ash

```diff
@@ -697,3 +697,120 @@ boolean LX_meatMaid()
 
 	return false;
 }
+
+string LX_getDesiredWorkshed(){
+	string currentWorkshed = get_property("auto_workshed").to_lower_case();
+	//return the actual item name in case a shorthand is used
+	switch(currentWorkshed)
+	{
+		case "model train set":
+		case "train":
+			return "model train set";
+		case "cold medicine cabinet":
+		case "cmc":
+			return "cold medicine cabinet";
+		case "diabolic pizza cube": //unsupported
+		case "pizza": //unsupported
+			return "diabolic pizza cube";
+		case "Asdon Martin keyfob":
+		case "asdon":
+			return "Asdon Martin keyfob";
+		case "portable mayo clinic":
+		case "mayo":
+			return "portable mayo clinic";
+		case "little geneticist dna-splicing lab":
+		case "dnalab":
+			return "little geneticist dna-splicing lab";
+		case "snow machine": //unsupported
+		case "spinning wheel": //unsupported
+		case "warbear auto-anvil": //unsupported
+		case "warbear chemistry lab": //unsupported
+		case "warbear high-efficiency still": //unsupported
+		case "warbear induction oven": //unsupported
+		case "warbear jackhammer drill press": //unsupported
+		case "warbear lp-rom burner": //unsupported
+		default:
+			// auto_workshed is invalid or none/false/whatever to say don't do this
+			return "auto";
+	}
+}
+
+boolean LX_setWorkshed(){
+	string desiredShed = LX_getDesiredWorkshed();
+	string existingShed = get_workshed();
+	string workshedChanged = get_property("_workshedItemUsed");
+
+	if (workshedChanged == "true") return false; //Don't even try if the workshed has already been changed once
+
+	//Check to make sure we can use the workshed item and that it isn't already in the campground. If already in campground, return false also
+	//These first 2 ifs are only used if something other than auto is specified
+	if (auto_is_valid(to_item(desiredShed)) && (contains_text(existingShed, desiredShed) == false) && (item_amount(to_item(desiredShed)) > 0))

```

Comment:
```suggestion
	if (desiredShed != $item[none] && auto_is_valid(desiredShed) && existingShed != desiredShed && item_amount(desiredShed) > 0)

```

Reply:
Obviously need similar updates below. Note I added a check that the desired is not `none` to ensure that setting has been properly set before trying to use it